test/expect/TestJit.test_peephole-different_size.expect

```diff
@@ -0,0 +1,5 @@
+graph(%0 : Double(1)
+      %1 : Double(2)) {
+  %2 : Double(1) = aten::type_as(%0, %1)
```

Comment:
Shouldn't the peephole remove this `aten::type_as`?

i.e. if we have
```
x = torch.randn(2, dtype=torch.double)
y = torch.randn(1, dtype=torch.double)
z = x.type_as(y)
```
the type_as is a no-op because x, y have the same "type" (dtype, layout, backend)

Reply:
That case is for tensor with different sizes. Something like:
x = torch.tensor([0.4], requires_grad=True)
y = torch.tensor([0.4, 0.5], requires_grad=True)
z = x.type_as(y)

Reply:
Right, but in that case, the type_as is unnecessary (z is the same as x)